newIDE/app/src/Utils/GDevelopServices/Shop.js

```diff
@@ -31,3 +31,56 @@ export const listListedPrivateAssetPacks = async (): Promise<
   const response = await client.get('/asset-pack');
   return response.data;
 };
+
+type Purchase = {|
+  id: string,
+  productId: string,
+  buyerId: string,
+  receiverId: string,
+  createdAt: string,
+  cancelledAt?: string,
+|};
+
+export const listUserPurchases = async (
+  getAuthorizationHeader: () => Promise<string>,
+  {
+    userId,
+    productType,
+    role,
+  }: {|
+    userId: string,
+    productType: 'asset-pack',
+    role: 'receiver' | 'buyer',
+  |}
+): Promise<Array<Purchase>> => {
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await client.get('/purchase', {
+    headers: { Authorization: authorizationHeader },
+    params: { userId, productType, role },
+  });
+  return response.data;
+};
+
+export const getAuthorizationTokenForPrivateAssets = async (
+  getAuthorizationHeader: () => Promise<string>,
+  {
+    userId,
+  }: {|
+    userId: string,
+  |}
+): Promise<string> => {
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await client.post(
+    '/asset-pack/action/authorize',
+    {},
+    {
+      headers: { Authorization: authorizationHeader },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const createAuthorizedUrl = (url: string, token: string): string => {
```

Comment:
I think we're promising too much with this function name. It sounds like a universal way to make an URL authorized. Sure the implementation is simple, but it works only for a specific convention, which works only in the realm of our Shop api.
In other words, name this:
```suggestion
export const createProductAuthorizedUrl = (url: string, token: string): string => {
```